src/components/Router/index.js

```diff
@@ -98,12 +100,18 @@ class CrossDeviceMobileRouter extends Component {
       return this.setError()
     }
     this.setState({token, steps, step, loading: false})
+    this.resetI18n(language)
     actions.setDocumentType(documentType)
   }
 
   setError = () =>
     this.setState({crossDeviceError: true, loading: false})
 
+  resetI18n = (language) => {
+    this.setState({i18n: setI18n(language)})
+    this.props.actions.setTranslations(this.state.i18n.phrases)
```

Comment:
Do we need to use both component state and the redux store?

Reply:
Yes, we have because we are passing the polyglot instance as a prop down to the children, while the redux store is only keeping the translated strings.

Reply:
Doesn't the polyglot instance have access to the translated strings? Why do you need them in the store?